lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -115,7 +116,7 @@ def eval_block
 
         def verify_message
           return true if @expected_message.nil?
-          values_match?(@expected_message, @actual_error.message)
+          values_match?(@expected_message, @actual_error.message.to_s)
```

Comment:
I see the problem now -- in `composable_spec.rb`, we have this:

``` ruby
RSpec::Matchers.define :matcher_using_surface_descriptions_in do |expected|
  match { false }
  failure_message { surface_descriptions_in(expected) }
end
```

`surface_descriptions_in` isn't necessarily meant to return a string (it's meant to return a data structure of the same shape), but the matcher protocol says `failure_message` should be a string.  So our custom matcher is written incorrectly.

That said, I wonder if we should add a `to_s` somewhere for situations like this...maybe [here](https://github.com/rspec/rspec-expectations/blob/f69cc918be1bdaccdb8f85877178c62d89439bdc/lib/rspec/expectations/fail_with.rb#L27)?

I'm fine with keeping the `raise_error` fix and spec as well -- it makes it a bit more robust, I think.